x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/shared/components/take_action_dropdown.tsx

```diff
@@ -163,6 +164,13 @@ export const TakeActionDropdown = memo(
       [dataAsNestedObject]
     );
 
+    const isAlertSourceEndpoint = useMemo(() => {
+      const eventModules = getOr([], 'kibana.alert.original_event.module', dataAsNestedObject);
+      const kinds = getOr([], 'kibana.alert.original_event.kind', dataAsNestedObject);
+
+      return eventModules.includes('endpoint') && kinds.includes('alert');
+    }, [dataAsNestedObject]);
```

Comment:
uses the same logic as Alerts table context menu:
https://github.com/elastic/kibana/blob/90677bf58fbab79b63ec15107d8343c8796275af/x-pack/solutions/security/plugins/security_solution/public/detections/components/alerts_table/timeline_actions/alert_context_menu.tsx#L128-L133

Reply:
@gergoabraham , Does it make sense to make it a common util so that tomorrow, if there are any changes to the logic, we do not face the similar inconsistency?